.github/actions/publish/action.yml

```diff
@@ -0,0 +1,292 @@
+            electron-builder \
+              --config="${ELECTRON_BUILDER_CONFIG}" \
+              --c.extraMetadata.analytics.sentry.token="$(cat < '${{ runner.temp }}/sentry_dsn')" \
+              --c.extraMetadata.name="${APPLICATION_NAME}" \
+              --c.extraMetadata.packageType="${target}" \
+              --c.extraMetadata.version="${APPLICATION_VERSION}" \
+              "--${ELECTRON_BUILDER_OS}" \
+              ${ARCHITECTURE_FLAGS}
+
+            rm "${ELECTRON_BUILDER_CONFIG}"
+
+            if [[ $target =~ appimage ]]; then
+              BUILD_DIRECTORY="$(pwd)/dist"
+              APPIMAGE_PATH="${BUILD_DIRECTORY}/${PRODUCT_NAME}-${APPLICATION_VERSION}-${ELECTRON_BUILDER_ARCHITECTURE}.AppImage"
+              APPIMAGE_ZIP_PATH="${BUILD_DIRECTORY}/${APPLICATION_NAME}-${APPLICATION_VERSION}-${ELECTRON_BUILDER_OS}-${ELECTRON_BUILDER_ARCHITECTURE}.zip"
+
+              zip_file \
+                -f "${APPIMAGE_PATH}" \
+                -s "${ELECTRON_BUILDER_OS}" \
+                -o "${APPIMAGE_ZIP_PATH}"
+            fi
+        done
+
+        echo "version=${APPLICATION_VERSION}" >> $GITHUB_OUTPUT
+
+      env:
+        # Apple notarization (afterSignHook.js)
+        XCODE_APP_LOADER_EMAIL: ${{ inputs.XCODE_APP_LOADER_EMAIL }}
+        XCODE_APP_LOADER_PASSWORD: ${{ fromJSON(inputs.secrets).CUSTOM_JOB_SECRET_6 }}
+        # https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks
+        # https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#about-workflow-runs-from-public-forks
+        CSC_FOR_PULL_REQUEST: true
+
+    - name: Rename release assets
+      shell: bash --noprofile --norc -eo pipefail -x {0}
+      run: |
+        set -ea
+
+        [[ '${{ inputs.VERBOSE }}' =~ on|On|Yes|yes|true|True ]] && set -x
+
+        # generalise release asset names (latest) to avoid rebuilding/renaming on finalize
+        find dist -type f -maxdepth 1 \
+          -name "*${{ steps.package_release.outputs.version }}*" \
+          -exec sh -c 'mv {} $(echo {} | sed 's/${{ steps.package_release.outputs.version }}/latest/g')' \;
+
+    # https://www.electron.build/auto-update.html#staged-rollouts
+    - name: Configure staged rollout(s)
+      shell: bash --noprofile --norc -eo pipefail -x {0}
+      run: |
+        set -ea
+
+        [[ '${{ inputs.VERBOSE }}' =~ on|On|Yes|yes|true|True ]] && set -x
+
+        percentage="$(cat < repo.yml | yq e .triggerNotification.stagingPercentage)"
+
+        find dist -type f -maxdepth 1 \
+          -name "latest*.yml" \
+          -exec yq -i e .version=\"${{ steps.package_release.outputs.version }}\" {} \;
+
+        find dist -type f -maxdepth 1 \
+          -name "latest*.yml" \
+          -exec yq -i e .stagingPercentage=\"$percentage\" {} \;
+
+    # https://github.com/softprops/action-gh-release#-customizing
+    - name: Create draft GitHub (pre)release
```

Comment:
@Page- @mcraa @builder555 @dfunckt the new approach is to create a draft release using the branch name and upload generalised (no version number in filenames) assets. This allows the PR to be built multiple times, rebuilt/rebased, etc. and not have the release polluted with multiple versions of assets potentially.

I am not sure if this is better than deleting the draft release and re-creating it with every PR run. I suppose either way will work, which way to you feel is better?